### ISSUE TITLE
While selecting multiple blocks, the page will crash once you go back to selecting only one block.

### ISSUE THREAD

**Issue by TimothyBJacobs** [WordPress/gutenberg#3275]
## Issue Overview
Changing the number of "blocks" selected works, but once you've broken out of a single block and are selecting multiple blocks, if you reduce your selection back to a single block, the app will crash.

- Safari 11.0
- Gutenberg 1.6
## Steps to Reproduce (for bugs)
1. Click into a block.
2. Hold down shift and the down arrow and start selecting.
3. Keep hitting the down arrow until the whole block is selected.
4. Hit the down arrow until multiple blocks are selected.
5. Switch to hitting the up arrow. 
6. When two blocks are selected, and you then hit the up arrow one more time, the page crashes.

This works going either up or down, as long as you go past one block and then go down to one again. 

```
[Error] TypeError: null is not an object (evaluating 'm.collapsed')
	performWork (react-dom.f04e56af.js:13950)
	scheduleUpdateImpl (react-dom.f04e56af.js:14256)
	enqueueSetState (react-dom.f04e56af.js:10717)
	setState (react.83acb8f5.js:454)
	onStateChange (index.js:18:20477)
	(anonymous function)
	notify (index.js:28:6565)
	notifyNestedSubs (index.js:28:7099)
	(anonymous function)
	onStateChange (index.js:18:20503)
	(anonymous function)
	u (index.js:18:22175)
	onMultiSelect (index.js:39:91049)
	value (index.js:39:89803)
	value (index.js:39:90318)
	value
```

## Expected Behavior
I expected the selection to go to the single block.

## Current Behavior
Page crashed.

**Comment by ephox-mogran** [comment]
Ah, it looks like I missed a guard when tweaking #3038 to handle backwards selections. We are having difficulty replicating this, because you need a situation where the selection range is null. But the fix should just be to check that `range` is defined in `dom.js`:

```
        if ( collapseRanges && ! range.collapsed ) {
		const newRange = document.createRange();
		// Get the end point of the selection (see focusNode vs. anchorNode)
		newRange.setStart( selection.focusNode, selection.focusOffset );
		newRange.collapse( true );
		range = newRange;
	}
```

Should be `collapseRanges && range && ! range.collapsed `. However, that's just a guess, as we aren't replicating it at the moment.

**Comment by ephox-mogran** [comment]
Actually, just pulled from `master` and I can definitely replicate it. Maybe it is something else. Investigating.

**Comment by ephox-mogran** [comment]
So what is happening is probably due to the recent change (#3253) where focus is being reset to the body on a multi-selection. Therefore the  `focus` prop is null, and the `focus.collapsed` check for showingUI in block is an error. I'll add the guard listed above as well, because it should be there, but it wasn't causing the problem **yet** :)

**Comment by nosolosw** [comment]
I experienced something similar (I actually think it's the same, but not sure why the console message would be different, so I better report it, just in case). Adding the capture in case it helps:

![block-selection-disappear](https://user-images.githubusercontent.com/583546/32265214-66c68a36-bee2-11e7-97a7-306a55b8c07a.gif)

What I did was:

* Click on a block.
* With SHIFT enabled, click the arrow down key: the result is that the blocks are selected.
* With SHIFT enabled, click the arrow up kep: the result is that the blocks are unselected until it reaches the first one.

The app crashes and the console show a `TypeError: Cannot read property "collapsed" of null`.
